src/ewatercycle/base/forcing.py

```diff
@@ -222,9 +222,16 @@ def save(self):
         # Copy shapefile so statistics like area can be derived
         if clone.shape is not None:
             if not clone.shape.is_relative_to(clone.directory):
-                clone.shape = Path(
+                new_shp_path = Path(
                     shutil.copy(clone.shape, clone.directory / clone.shape.name)
                 )
+                # Also copy other required files:
+                for ext in [".dbf", ".shx", ".prj"]:
```

Comment:
The '.prj' file is optional. Can you check that file exists before copying it?

If I run 

```python
from ewatercycle.testing.fixtures import rhine_shape
from ewatercycle.forcing import sources
from pathlib import Path
import shutil
fc = sources['CaravanForcing']
Path('/tmp/shptest').mkdir(exist_ok=True)
s = rhine_shape()
shutil.copy(s, '/tmp/shptest/rhine.shp')
shutil.copy(s.with_suffix('.dbf'), '/tmp/shptest/')
shutil.copy(s.with_suffix('.shx'), '/tmp/shptest/')
fd = Path('/tmp/cf')
fd.mkdir()
f = fc.generate(start_time="1991-01-01T00:00:00Z", end_time="1991-12-31T00:00:00Z", shape='/tmp/shptest/rhine.shp', directory=fd, basin_id='camels_01022500')
```

I get
```python
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[5], [line 1](vscode-notebook-cell:?execution_count=5&line=1)
----> [1](vscode-notebook-cell:?execution_count=5&line=1) f = fc.generate(start_time="1991-01-01T00:00:00Z", end_time="1991-12-31T00:00:00Z", shape='/tmp/shptest/Rhine.shp', directory=fd, basin_id='camels_01022500')

File ~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:244, in CaravanForcing.generate(cls, start_time, end_time, directory, variables, shape, **kwargs)
    [229](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:229)     ds_basin_time[var].to_netcdf(
    [230](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:230)         Path(directory)
    [231](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:231)         / f"{basin_id}_{start_time_name}_{end_time_name}_{var}.nc"
    [232](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:232)     )
    [234](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:234) forcing = cls(
    [235](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:235)     directory=Path(directory),
    [236](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:236)     start_time=start_time,
   (...)
    [242](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:242)     },
    [243](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:243) )
--> [244](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:244) forcing.save()
    [245](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/_forcings/caravan.py:245) return forcing

File ~/git/eWaterCycle/ewatercycle/src/ewatercycle/base/forcing.py:230, in DefaultForcing.save(self)
    [228](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/base/forcing.py:228)     # Also copy other required files:
    [229](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/base/forcing.py:229)     for ext in [".dbf", ".shx", ".prj"]:
--> [230](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/base/forcing.py:230)         shutil.copy(
    [231](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/base/forcing.py:231)             clone.shape.with_suffix(ext),
    [232](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/git/eWaterCycle/ewatercycle/src/ewatercycle/base/forcing.py:232)             clone.directory / clone.shape.with_suffix(ext).name,
...
    [261](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/miniforge3/envs/ewatercycle/lib/python3.12/shutil.py:261)         try:
    [262](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/miniforge3/envs/ewatercycle/lib/python3.12/shutil.py:262)             with open(dst, 'wb') as fdst:
    [263](https://file+.vscode-resource.vscode-cdn.net/home/verhoes/git/eWaterCycle/ewatercycle/~/miniforge3/envs/ewatercycle/lib/python3.12/shutil.py:263)                 # macOS

FileNotFoundError: [Errno 2] No such file or directory: '/tmp/shptest/Rhine.prj'
```

Reply:
.prj files are required for our workflow, as we cannot reliably guess what projection the shapefile is in (probably WGS-84, but who knows). If we guess the projection incorrectly the user will get incorrect results without ever knowing (clearly) why.

Therefore I think it's best to keep it as mandatory. I will add a more clear error.